### PR TITLE
Fix deadline behavior

### LINKED ISSUE #26: Deadline should take a uint64 timestamp not uint32 seconds
- Add test to catch this as well

### PR DESCRIPTION
**Pull request by alexkeating**
### Description

- Deadline behavior did not make any sense. We need to take in a timestamp not a number of seconds added on to a timestamp.

Closes #26

### CODE REVIEW

File: test/PerpetualPositionRouter.t.sol [hunk 1]
```diff
@@ -31,7 +31,7 @@ contract PositionRouterTest is Test, PerpetualContracts {
       abi.encodePacked(
         uint8(openFunc),
         uint160(sqrtPriceLimitX96),
-        uint32(type(uint32).max),
+        uint32(block.timestamp),
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:34]: Why did you change all the test timestamps? Both values should be fine, just curious

File: test/PerpetualPositionRouter.t.sol [hunk 2]
```diff
@@ -221,22 +221,35 @@ contract Fallback is PositionRouterTest {
...
     assertTrue(!ok);
   }
 
   function testFork_FailedFallbackWithZeroFuncId() public {
+    delegateApproval.approve(vethPositionRouterAddr, 1);
     (bool ok,) = payable(vethPositionRouterAddr).call(
-      abi.encodePacked(uint8(0), uint160(0), uint32(type(uint32).max), uint96(1 ether), uint96(0))
+      abi.encodePacked(uint8(0), uint160(0), uint32(block.timestamp), uint96(1 ether), uint96(0))
+    );
+    assertTrue(!ok);
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:244]: Looks like we have this in a few places
```suggestion
    assertFalse(ok);
```

File: test/PerpetualPositionRouter.t.sol [hunk 3]
```diff
@@ -221,22 +221,35 @@ contract Fallback is PositionRouterTest {
...
   function testFork_FailedClosePositionCallWithWrongArguments() public {
+    delegateApproval.approve(vethPositionRouterAddr, 1);
     (bool ok,) = payable(vethPositionRouterAddr).call(
-      abi.encodePacked(uint8(5), uint160(0), uint32(type(uint32).max), uint96(1 ether), uint96(0))
+      abi.encodePacked(uint8(5), uint160(0), uint32(block.timestamp), uint96(1 ether), uint96(0))
     );
     assertTrue(!ok);
   }
 
   function testFork_FailedFallbackWithZeroFuncId() public {
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:239]: the naming condition for tests expecting a revert should be (can do this in a separate PR or this one)
```suggestion
  function testFork_RevertIf_FallbackWithZeroFuncId() public {
```

File: test/PerpetualPositionRouter.t.sol [hunk 4]
```diff
@@ -221,22 +221,35 @@ contract Fallback is PositionRouterTest {
   }
 
   function testFork_FailedCallWhenExtraCalldataArgument() public {
+    delegateApproval.approve(vethPositionRouterAddr, 1);
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:224]: Why was this line added to some tests, seems they passed properly both with and without it?

File: test/PerpetualPositionRouter.t.sol (hunk 1 above)
Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol:34]: The idea was that this would be closer a more realistic value and would be more likely to catch a regression.

File: test/PerpetualPositionRouter.t.sol (hunk 4 above)
Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol:224]: These checks are checking for failures and if the delegate has not approved the router than the call will fail. I added it to make sure the other conditions are causing a failure. Is there a better way when calling fallback to get the specific error?

Comment (mds1) [reply test/PerpetualPositionRouter.t.sol:224]: Ahh you're saying that because we're not using `expectRevert` since it's a low level call, it's possible these were bad tests, in that they were failing due to a lack of approval.

We can do this in a follow up PR, but we should also be checking the return data to make sure the revert reason matches what we expect

File: test/PerpetualPositionRouter.t.sol (hunk 3 above)
Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol:239]: Created a followup issue https://github.com/ScopeLift/optimizoors-perpetual-protocol/issues/29

File: test/PerpetualPositionRouter.t.sol (hunk 2 above)
Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol]: https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/27/commits/f74d1ce6801af48d966d018187206af5c0839a69

**Review by mds1 — APPROVED** [review]
